### PR TITLE
TST: update .travis.yml

### PR DESCRIPTION
**Pull request by rgommers**
I turned on TravisCI for this repo. Would be good to have it working before merging other PRs.

Changes in this PR are basically a subset of relevant changes to ``scipy/.travis.yml``.

### CODE REVIEW

**Comment by pv** [comment]
Maybe replace `scipy.lib.six` by `six` and add that as dependency?
Maybe specify `sudo: false` and use [the addons section instead of apt-get](https://github.com/scipy/scipy/blob/master/.travis.yml#L64) for installing the packages.

**Comment by rgommers** [comment]
Fine with me to add ``six`` as a dependency. Could be vendored in this repo, but it's a very light dependency.

**Comment by rgommers** [comment]
Or even better, just replace the import with:

    PY3 = sys.version_info[0] == 3
    if PY3:
        string_types = str
        import builtins.open as _open
    else:
        string_types = basestring,
        import __builtin__.open as _open

Then no dependency on ``six`` or ``scipy._lib.six`` is needed.

**Comment by rgommers** [comment]
> Maybe specify sudo: false and use the addons section instead of apt-get for installing the packages.

Adding ``sudo: false``. I don't get the addons section - I am doing that in this PR already right?

File: weave/_dumbdbm_patched.py [hunk 1]
```diff
@@ -22,12 +22,17 @@
...
 
-from scipy.lib.six import builtins
-from scipy.lib.six import string_types
 
-_open = builtins.open
+PY3 = sys.version_info[0] == 3
+if PY3:
+    string_types = str
+else:
+    string_types = basestring
```
Comment (pv) [inline weave/_dumbdbm_patched.py]: It's actually a tuple:
```
if PY3:
    string_types = str,
else:
    string_types = basestring,
```

Comment (rgommers) [inline weave/_dumbdbm_patched.py]: thanks, fixed

**Comment by pv** [comment]
Otherwise LGTM

**Comment by rgommers** [comment]
tests passed

**Comment by rgommers** [comment]
Okay, merging then. Thanks @pv

**Pull request merged**